### PR TITLE
Add search filter to the admin menu player tab

### PR DESCRIPTION
**Pull request by ShadowCommander**
## About the PR

Adds a search bar filter that searches player name, character name, character identity, and character job.

## Technical details

Implements a new control called SearchListContainer that adds search functionality to a ListContainer.

Reworks the player tab to use SearchListContainer.

Fix PlayerListControl handling when the selected button is deleted.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/10494922/a16f1e3d-44ad-4025-b902-87dd9d9f38a3

## Breaking changes

**Changelog**

Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl:
ADMIN:
- add: Added a search bar to the admin menu player tab.
- tweak: Changed the player tab entry tooltip to be the player name, character name, identity, and job. Moved the time tooltip to the header.

### CODE REVIEW

**Review by juliangiebel — APPROVED** [review]
Lgtm.

File: Content.Client/Administration/UI/Tabs/PlayerTab/PlayerTab.xaml.cs [hunk 1]
```diff
@@ -108,29 +115,76 @@ private void RefreshPlayerList(IReadOnlyList<PlayerInfo> players)
...
             var useAltColor = false;
-            foreach (var player in sortedPlayers)
+            var entry = new PlayerTabEntry(player.Username,
+                player.CharacterName,
+                player.IdentityName,
+                player.StartingJob,
+                player.Antag ? "YES" : "NO",
+                new StyleBoxFlat(useAltColor ? _altColor : _defaultColor),
+                player.Connected,
+                player.PlaytimeString);
```
[the remark this answers is not in the document]
Comment (ShadowCommander) [reply Content.Client/Administration/UI/Tabs/PlayerTab/PlayerTab.xaml.cs]: Yes. Moved it to inside the constructor in [1f165ed](https://github.com/space-wizards/space-station-14/pull/28030/commits/1f165edf87df04ef372f766b2190cf16173b635e)

File: Content.Client/Administration/UI/Bwoink/BwoinkWindow.xaml.cs [hunk 2]
```diff
@@ -16,14 +16,17 @@ public BwoinkWindow()
 
             Bwoink.ChannelSelector.OnSelectionChanged += sel =>
             {
-                if (sel is not null)
+                if (sel is null)
                 {
-                    Title = $"{sel.CharacterName} / {sel.Username}";
+                    Title = "None selected";
```
Comment (AJCM-git) [inline Content.Client/Administration/UI/Bwoink/BwoinkWindow.xaml.cs:21]: GetString

File: Content.Client/Administration/UI/Tabs/PlayerTab/PlayerTabEntry.xaml.cs [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
-        if (identity != character)
-            CharacterLabel.Text += $" [{identity}]";
-        AntagonistLabel.Text = antagonist;
-        BackgroundColorPanel.PanelOverride = styleBox;
-        OverallPlaytimeLabel.Text = overallPlaytime;
+        JobLabel.Text = player.StartingJob;
+        CharacterLabel.Text = player.CharacterName;
+        if (player.IdentityName != player.CharacterName)
+            CharacterLabel.Text += $" [{player.IdentityName}]";
+        AntagonistLabel.Text = player.Antag ? "YES" : "NO";
```
Comment (AJCM-git) [inline Content.Client/Administration/UI/Tabs/PlayerTab/PlayerTabEntry.xaml.cs:25]: same

**Review by AJCM-git — APPROVED** [review]
Need to do a more through dive but this looks good

File: Content.Client/Administration/UI/Bwoink/BwoinkWindow.xaml.cs (hunk 2 above)
Comment (ShadowCommander) [reply Content.Client/Administration/UI/Bwoink/BwoinkWindow.xaml.cs]: Done

**Pull request merged**